bundle/Command/UpdateContentVisibilityCommand.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Command;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Language;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Configuration\ScheduledVisibilityConfiguration;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\VisibilityUpdateResult;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Exception\InvalidStateException;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Service\ScheduledVisibilityService;
+use Pagerfanta\Doctrine\DBAL\QueryAdapter;
+use Pagerfanta\Pagerfanta;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use function count;
+use function sprintf;
+
+final class UpdateContentVisibilityCommand extends Command
+{
+    private SymfonyStyle $style;
+
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ScheduledVisibilityService $scheduledVisibilityService,
+        private readonly ScheduledVisibilityConfiguration $configurationService,
+        private readonly Connection $connection,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'Updates content visibility based on publish_from and publish_to attributes and configuration.',
+        );
+        $this->addOption(
+            'limit',
+            'l',
+            InputOption::VALUE_OPTIONAL,
+            'Number of content object to process in a single iteration',
+            50,
+        );
+        $this->addOption(
+            'ttl',
+            't',
+            InputOption::VALUE_OPTIONAL,
+            'Expiration time for language caching',
+            3600,
+        );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->style = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->style->info(
+            'This command fetches content and updates visibility based on its schedule from publish_from and publish_to fields.',
+        );
+
```

Comment:
We can add a confirmation question here with:

```php
$question = new ConfirmationQuestion(
    'Continue with this action?',
    false,
    '/^(y)/i',
);

if (!$this->symfonyStyle->askQuestion($question)) {
    $this->symfonyStyle->success('Aborted');

    return Command::SUCCESS;
}
```

Then, when configuring a cronjob, we'll just add `--no-interaction` option to skip it.

Reply:
fixed in b325c484488f68c043f8249a60b0d515697f35f5

Reply:
Looks good, but it seems that the default value for the question needs to be `true` in order for `--no-interaction` to pass it.

Reply:
fixed in 8dee80c4933afd831430ff6bdc6d76072a9cff66